pkg/provider/azure.go

```diff
@@ -382,7 +386,12 @@ func (az *Cloud) InitializeCloudFromConfig(ctx context.Context, config *config.C
 		var authProvider *azclient.AuthProvider
 		authProvider, err = azclient.NewAuthProvider(&az.ARMClientConfig, &az.AzureClientConfig.AzureAuthConfig)
 		if err != nil {
-			return err
+			if errors.Is(err, azclient.ErrNoValidAuthMethodFound) {
```

Comment:
please note: Will skip this error. @nilo19 @feiskyer @MartinForReal

Reply:
Need update after https://github.com/kubernetes-sigs/cloud-provider-azure/pull/8730.